server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -75,6 +76,7 @@
   private final AvaticaServerConfiguration config;
   private final Subject subject;
   private final SslContextFactory sslFactory;
+  private final List<ServerCustomizer> serverCustomizers;
```

Comment:
I think we should force this to be non-null (as you did in the Builder). An empty list as the "default" value passed in through the constructors would let us eliminate the if-null check elsewhere.